pallets/ethereum-events/src/lib.rs

```diff
@@ -1641,6 +1644,30 @@ impl<T: Config> ProcessedEventsChecker for Pallet<T> {
         <ProcessedEvents<T>>::insert(event_id.clone(), accepted);
         Ok(())
     }
+
+    fn migrate_processed_events_batch() -> Option<BoundedVec<EventMigration, ProcessingBatchBound>>
+    {
+        let batch_size: u32 = ProcessingBatchBound::get();
+        let entry_return =
+            |event: &EthEventId, outcome: bool| ProcessedEvents::<T>::insert(event, outcome);
+        let migration_batch: Vec<EventMigration> = ProcessedEvents::<T>::iter()
+            .take(batch_size as usize)
+            .map(|(event_id, outcome)| EventMigration {
+                event_id,
+                outcome,
+                entry_return_impl: entry_return,
+            })
+            .collect();
+
+        migration_batch.iter().for_each(|event_to_migrate| {
```

Comment:
can we avoid doing 2 iterations here? Is there a way to drain/take a `batch_size` number of events?

Reply:
excellent question. There is a way in a double map, but not a single map.
Ultimately the count of reads and writes in the storage is the same.